lib/internal/timers.js

```diff
@@ -173,6 +174,14 @@ class Timeout {
                             ' a 32-bit signed integer.' +
                             '\nTimeout duration was set to 1.',
                             'TimeoutOverflowWarning');
+      } else if (after < 0) {
+        process.emitWarning(`${after} is a negative number.` +
```

Comment:
This likely doesn't need to email a warning on every instance, that would likely get a bit spammy. Perhaps this can emit once per process?

Reply:
I personally think passing invalid delay (negative number / NaN) is usually unintentional by the user, with the warning per instance they can leverage the `--trace-warnings` flag to find out where it was exactly called and fix them accordingly, happy to discuss more!

Reply:
Hi James @jasnell just trying to follow up to see if you are still happy with the change. If you think "emit once per process" would make more senses, would you mind pointing out if I am on the right track.

_I may be completely wrong_ but here is my thought:
1. create an event emitter
2. use `emitOnce` only if the condition is reached inside `else if`
3. in the callback use `process.emitWarning`

Thanks 🙏

Reply:
I still don't believe it needs to emit a warning on every instance.

Reply:
I think we can take a look at the pros and cons side by side from my point of view @jasnell and others:

**Pros:**
* Align with `TimeoutOverflowWarning` which is also emiting a warning on every instance [(ref)](https://github.com/nodejs/node/blob/main/lib/internal/timers.js#L172).
* This would help the user easily find out where it is being called


**Cons:**
* Make the console a bit noisy

My thought is that we should either keep the current behaviour or we should also change `TimeoutOverflowWarning` to only emit once per process.

Reply:
I found a legacy script we used to use at work which reads the delay from env variable and pass into the `setTimeout` and `setInterval` - because it was a string and we didn't use schema validator at the time, the delay was always set to `1` without knowing it.

Had warning trace to tell us which one was not a number would be extremely helpful.

Reply:
Fixed, both negative number and NaN passed to timer will only emit warning once per process